include/seqan3/core/debug_stream/range.hpp

```diff
@@ -68,80 +69,162 @@ constexpr bool reference_type_is_streamable_v<rng_t, char_t> = true;
+        requires std::same_as<std::remove_cvref_t<range_t>, std::remove_cvref_t<rng_t>>
+    constexpr void operator()(stream_t & stream, range_t && r) const
     {
-        s << *b;
-        ++b;
+        stream << '[';
+        auto first = std::ranges::begin(r);
+        auto last = std::ranges::end(r);
+        if (first != last)
+        {
+            stream << *first;
+            ++first;
+        }
+        while (first != last)
+        {
+            stream << ',';
+            stream << *first;
+            ++first;
+        }
+        stream << ']';
     }
-    while (b != e)
+};
+
+/*!
+ * \brief A printer for (biological) sequences.
+ *
+ * The (biological) sequence (except for ranges over unsigned integers) is printed just as if it were a string, i.e.
+ * <tt>std::vector<dna4>{'C'_dna4, 'G'_dna4, 'A'_dna4}</tt> is printed as "CGA".
+ *
+ * \tparam sequence_t The type of the sequence to be printed; must model seqan3::sequence
+ * \ingroup core_debug_stream
+ */
+template <typename sequence_t>
+    requires sequence<sequence_t>
+struct sequence_printer<sequence_t>
+{
+    /*!
+     * \brief Prints the elements of a sequence to an output stream.
+     *
+     * \tparam stream_t The type of the output stream.
+     * \tparam range_t The type of the sequence to be printed; must be the same type as `sequence_t`.
+     *
+     * \param[in,out] stream The output stream to print to.
+     * \param[in] sequence The sequence to be printed.
+     */
+    template <typename stream_t, typename range_t>
+        requires std::same_as<std::remove_cvref_t<range_t>, std::remove_cvref_t<sequence_t>>
+    constexpr void operator()(stream_t & stream, range_t && sequence) const
     {
-        s << ',';
-        s << *b;
-        ++b;
+        for (auto && chr : sequence)
+            stream << chr;
     }
-    s << ']';
-
-    return s;
-}
+};
 
-/*!\brief All biological sequences can be printed to the seqan3::debug_stream.
- * \tparam sequence_t Type of the (biological) sequence to be printed; must model seqan3::sequence.
- * \param s The seqan3::debug_stream.
- * \param sequence The input range.
- * \relates seqan3::debug_stream_type
+/*!
+ * \brief Checks if a type is a character type.
  *
- * \details
+ * Checks whether `type` is a character type. Evaluates to `true`, if `type` is the type `char`, `char8_t`, `char16_t`,
+ * `char32_t`, or `wchar_t`, including cv-qualified variants. Otherwise, evaluates to `false`.
  *
- * The (biological) sequence (except for ranges over unsigned integers) is printed just as if it were a string, i.e.
- * <tt>std::vector<dna4>{'C'_dna4, 'G'_dna4, 'A'_dna4}</tt> is printed as "CGA".
+ * \tparam type The type to check.
+ * \ingroup core_debug_stream
+ */
+template <typename type>
+static constexpr bool is_char_type_v =
+    std::same_as<type, char> || std::same_as<type, char8_t> || std::same_as<type, char16_t>
+    || std::same_as<type, char32_t> || std::same_as<type, wchar_t>;
```

Comment:
`include/seqan3/alphabet/adaptation/char.hpp` has a concept

```cpp
template <typename type>
constexpr bool is_char_adaptation_v = std::same_as<type, char> || std::same_as<type, char16_t>
                                   || std::same_as<type, char32_t> || std::same_as<type, wchar_t>;
```

(`char8_t` missing seems like an oversight)

Reply:
Yes. So it should be replaced with the concept and the concept needs to be updated to add char8_t, which is a distinct type (not the same as char).
The last will be done in a separate PR